### PR TITLE
Add support for component template APIs

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Adds support for the component template APIs

### Issues Resolved
Part of #404 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Xtansia** [comment]
@dblock @Yury-Fridlyand @VachaShah Could I get a review on this please?

**Review by dblock — DISMISSED** [review]
LGTM, only did a cursory look

**Comment by Xtansia** [comment]
@dblock Had to rebase and fix conflicts, could I get a re-approval please?

File: tests/Tests/Cluster/ComponentTemplate/ComponentTemplateCrudTests.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+/*
+* Modifications Copyright OpenSearch Contributors. See
+* GitHub history for details.
+*
+*  Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (VachaShah) [inline tests/Tests/Cluster/ComponentTemplate/ComponentTemplateCrudTests.cs:11]: Does this license need to be here?

**Review by VachaShah — COMMENTED** [review]
@Xtansia Is it not possible to generate these APIs? Or are they already generated?

File: tests/Tests/Cluster/ComponentTemplate/ComponentTemplateCrudTests.cs (hunk 1 above)
Comment (Xtansia) [reply tests/Tests/Cluster/ComponentTemplate/ComponentTemplateCrudTests.cs]: Have removed it

**Comment by Xtansia** [comment]
> @Xtansia Is it not possible to generate these APIs? Or are they already generated?

@VachaShah They're partially generated, the methods on the client class and the parts of the objects defining query parameters are generated (they're under a `_Generated` path). It's just the req/resp bodies and tests that are hand written at the moment.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**